### PR TITLE
fix Tangent ONB and DiagONB and add a Test.

### LINKED ISSUE #457: `DefaultOrthonormalBasis` on SPD(3) is not orthonormal
Here is a short-ish reproducer:
```julia
julia> M = SymmetricPositiveDefinite(3)
SymmetricPositiveDefinite(3)

julia> q = [2.0 0.0 0.0; 0.0 2.0 0.0; 0.0 0.0 1]
3×3 Matrix{Float64}:
 2.0  0.0  0.0
 0.0  2.0  0.0
 0.0  0.0  1.0

julia> b = get_basis(M, q, DefaultOrthonormalBasis())
DefaultOrthonormalBasis(ℝ) with 6 basis vectors:
 E1 =
  3×3 Matrix{Float64}:
   1.0  0.0  0.0
   0.0  0.0  0.0
   0.0  0.0  0.0
 E2 =
  3×3 Matrix{Float64}:
   0.0       0.707107  0.0
   0.707107  0.0       0.0
   0.0       0.0       0.0
 ⋮
 E5 =
  3×3 Matrix{Float64}:
   0.0  0.0       0.0
   0.0  0.0       0.707107
   0.0  0.707107  0.0
 E6 =
  3×3 Matrix{Float64}:
   0.0  0.0  0.0
   0.0  0.0  0.0
   0.0  0.0  1.0

julia> norm(M, q, b.data[1])
0.4999999999999999
```

### PR DESCRIPTION
**Pull request by kellertuer**
This will fix #457.

Main Changes: We have to take the Eigenvectors of p^-1/2 V p^-1/2 (if V is the frame direction, V = I for ONB so then just p^-1) and the ONB vectors have to be post multiplied with p^1/2 * Xi * p^1/2

* [x] we have to adapt get_vector
* [x] we have to adapt get_coordinates
* [x] we should document this in the docstrings 

But we still need to adapt get_vector and get_coordinates since they are nor much more complicated.

### CODE REVIEW

**Comment by kellertuer** [comment]
My local pycall broke again, so I have not yet seen the docs, but the vector/coordinates are adapted and documented (just sat down simplified both a little bit). Take a look whether they can be made even faster.

I hope all three docs are understandable.

**Comment by kellertuer** [comment]
HM then this was also never covered by tests? Interesting, I will have to take a much closer look than, what I basically did was (for coordinates) simplify the many inner products and ah, I see, I am not allowed to do what I did for get vector the p^1/2 have to be within the sum (or I can only factor out one of them).

**Comment by mateuszbaran** [comment]
We probably only tested `get_vector` and `get_coordinates` at identity matrix, so the tests were quite easy to pass.

**Comment by kellertuer** [comment]
Sure, I added your example as a test, but I will need some time with pen&paper again to check that both my derivations / simplifications for vector and coordinates are correct.

**Comment by kellertuer** [comment]
The last commit is code that works, but the ad hoc generation of the basis vectors and the `inner` could be simplified still (that is what I tried before but must have made a mistake in rearranging the `tr`.

get_vector is finished and works, we can leave `get_coordinates` as it is currently, but there is also space for more performance.

**Comment by mateuszbaran** [comment]
Thanks! I'll check it soon.

**Comment by mateuszbaran** [comment]
I've significantly improved performance. I get roughly 500 ns for get_coordinates and 200 ns for get_vector, on static arrays. There are no obvious optimizations for these two functions. get_basis may be still worth optimizing in a similar way (mainly updating `(V[:, i] * transpose(V[:, j]) + V[:, j] * transpose(V[:, i]))` in-place).

**Comment by kellertuer** [comment]
Great, thanks for the ideas, not only did they make the code faster, it is also more readable now. I adopted that for the basis as well.

File: src/manifolds/SymmetricPositiveDefiniteLinearAffine.jl [hunk 1]
```diff
@@ -117,64 +117,182 @@ Return a orthonormal basis `Ξ` as a vector of tangent vectors (of length
...
+@doc raw"""
+    get_vector(::SymmetricPositiveDefinite, p, c, ::DefaultOrthonormalBasis)
+
+Using the basis from [`get_basis`](@ref  get_basis(M::SymmetricPositiveDefinite,p,B::DefaultOrthonormalBasis{<:Any,ManifoldsBase.TangentSpaceType}))
+the vector reconstruction with respect to this ONB can be simplified to
+
+```math
+   X = p^{\frac{1}{2}} \Biggl( \sum_{i=1,j=i}^n c_k \Delta_{i,j} \Biggr) p^{\frac{1}{2}}
+```
+where $k$ is trhe linearized index of the $i=1,\ldots,n, j=i,\ldots,n$.
```
Comment (mateuszbaran) [inline src/manifolds/SymmetricPositiveDefiniteLinearAffine.jl:262]:
```suggestion
where $k$ is the linearized index of the $i=1,\ldots,n, j=i,\ldots,n$.
```

File: src/manifolds/SymmetricPositiveDefiniteLinearAffine.jl [hunk 2]
```diff
@@ -117,64 +117,182 @@ Return a orthonormal basis `Ξ` as a vector of tangent vectors (of length
...
-    @assert size(X) == (div(N * (N + 1), 2),)
-    @assert size(Y) == (N, N)
+    @assert size(c) == (div(N * (N + 1), 2),)
+    @assert size(X) == (N, N)
+    e = eigen(Symmetric(p))
+    U = e.vectors
+    S = max.(e.values, floatmin(eltype(e.values)))
+    Ssqrt = Diagonal(sqrt.(S))
+    pSqrt = Symmetric(U * Ssqrt * transpose(U))
+    #V = Matrix{Float64}(I, N, N)
```
Comment (mateuszbaran) [inline src/manifolds/SymmetricPositiveDefiniteLinearAffine.jl:280]:
```suggestion
```

**Comment by mateuszbaran** [comment]
Great, I think `DiagonalizingOrthonormalBasis` could be modified in the same way as well?

**Comment by mateuszbaran** [comment]
... or maybe not, `V` doesn't have to be diagonal there so it's less clear.

**Review by mateuszbaran — APPROVED** [review]

**Comment by kellertuer** [comment]
no maybe not that easy because we do not necessarily have unit vectors but really have to form the corresponding V'V matrices.

**Comment by mateuszbaran** [comment]
Could we maybe bump version, merge this and tag a new release? There doesn't seem anything left to be done here.

**Comment by kellertuer** [comment]
Sure, no problem.

**Pull request merged**